### DISCUSSION TITLE
How to Set Up and Run Anuvaad Completely on Local Machine?

### DISCUSSION THREAD

**Discussion by mahendra-shah** [project-anuvaad/anuvaad#945] [Q&A] [answered]
Hello Team,

I hope this message finds you well.

I am currently attempting to run **Anuvaad** on my local machine to test its full functionality. I have successfully followed the documentation and managed just to run the dashboard (`anuvaad/anuvaad-fe/anuvaad-dashboard`) using **Node v18.20.3**, as well as the user management server (`anuvaad/anuvaad-api/anuvaad-user-management/user-management/user-management`).

However, I am encountering significant difficulties in setting up and running the entire project, including the database. As I am new to open source, I am unsure how to proceed with the complete setup to get everything up and running on my local environment to test all the features.

**Could someone please provide me with a step-by-step guide or any additional resources to help me set up the entire Anuvaad project locally?** Any advice or support from those who have successfully navigated this setup would be invaluable to me.

A special thanks to @aravinth for redirecting me here for assistance.

Thank you all in advance for your time and help.

**Comment by aravinth** [comment, marked as answer]
Hi @mahendra-shah . Thanks for your interest.
Can you please confirm if you had already gone through the documentation  : 

Two useful links under it are : 
1.  https://anuvaad.sunbird.org/modules/modulewise-appendix
2. https://anuvaad.sunbird.org/engage/kt-videos

Once you go through the technical KT videos of the modules, you can highlight which feature you want to work on. 
@aswinpradeep  - Can you please keep an eye on this thread?

**Comment by bhupend-deshmukh** [comment]
Hello Team,

I’m currently working on setting up the anuvaad/anuvaad-api/anuvaad-user-management/user-management/user-management module in my local environment. Unfortunately, I am facing several issues related to the database setup.

I have followed the documentation closely and also referenced the resources provided in this discussion. Despite these efforts and following up on all the resources shared, I am still encountering problems that prevent me from successfully completing the setup.

Specific issues I'm facing:

Database Configuration: Difficulty in configuring the database and tables as per the documentation.
Error Messages: Encountering specific error messages that are not resolved by the usual troubleshooting steps.
Setup Process: Unclear on certain steps in the setup process that might be causing these issues.
Could anyone please provide additional guidance or suggest any alternative resources that might help? Any insights or solutions to these problems would be greatly appreciated.

Thank you for your support!

**Reply by aswinpradeep** [reply]
Hi Bhupend, thanks for expressing interest in contributing.
1. Ensure DB is properly running, and just place the correct connection string.
  This is how we provide it as environment variable : MONGO_CLUSTER_URL=mongodb://xxx.xx.xx.xx:27017

2. Did you try running the docker image? The only error i could image will be caused by some versioning in packages by dependabot. You will be able to fix it by changing the docker base image to a newer one. Please provide more traces of the error in case you are not able to fix it. 

Additionaly, you can find the docker image of user management we run in testing env [here](https://hub.docker.com/layers/anuvaadio/anuvaad-user-management/231-c8c61cdc2/images/sha256-ecb545d28ed29e7cbbfc74c1b04258a65168f47c5ca9387cfb79c010198a6faf?context=explore). you may use the same

**Reply by bhupend-deshmukh** [reply]
Hi Aswin,

Thank you for the resources and guidance provided. I followed the steps as outlined, including pulling the Docker image and starting the server. However, I'm still encountering some issues after completing the setup.

Despite my efforts, the issue persists, and I'm finding it challenging to resolve the errors. I'd appreciate it if you could take a look at the details below and provide any additional insights or suggestions that could help me overcome this.

Looking forward to your guidance to help me get this project fully up and running.

Thank you for your support!


```
Traceback (most recent call last):
  File "/app/utilities/orgutils.py", line 32, in validate_org
    if result.count() == 0:
  File "/usr/local/lib/python3.8/site-packages/pymongo/cursor.py", line 822, in count
    return self.__collection._count(
  File "/usr/local/lib/python3.8/site-packages/pymongo/collection.py", line 1664, in _count
    return self.__database.client._retryable_read(
  File "/usr/local/lib/python3.8/site-packages/pymongo/mongo_client.py", line 1460, in _retryable_read
    server = self._select_server(
  File "/usr/local/lib/python3.8/site-packages/pymongo/mongo_client.py", line 1278, in _select_server
    server = topology.select_server(server_selector)
  File "/usr/local/lib/python3.8/site-packages/pymongo/topology.py", line 241, in select_server
    return random.choice(self.select_servers(selector,
  File "/usr/local/lib/python3.8/site-packages/pymongo/topology.py", line 199, in select_servers
    server_descriptions = self._select_servers_loop(
  File "/usr/local/lib/python3.8/site-packages/pymongo/topology.py", line 215, in _select_servers_loop
    raise ServerSelectionTimeoutError(
pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [Errno 111] Connection refused, Timeout: 30s, Topology Description: <TopologyDescription id: 66d9a42d9489aabd2abacc2b, topology_type: Single, servers: [<ServerDescription ('localhost', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('localhost:27017: [Errno 111] Connection refused')>]>
```
